staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go

```diff
@@ -1149,6 +1149,9 @@ type ManagedFieldsEntry struct {
 	// FieldsV1 holds the first JSON version format as described in the "FieldsV1" type.
 	// +optional
 	FieldsV1 *FieldsV1 `json:"fieldsV1,omitempty" protobuf:"bytes,7,opt,name=fieldsV1"`
+
+	// Subresource is the name of the subresource when the entry belongs to a subresource object.
```

Comment:
an update here to explain how without this field an update to spec and an update to status causes pathological behavior would be a big help

Reply:
Thanks @deads2k. I was thinking to change the comment to:

```
Subresource is the name of the subresource when the entry belongs to a subresource object. This field allows the same manager to change a subresource or spec and to treat these operations as distinct workflows. In other words, a change to a subresource and a change to spec using the same managers will be considered as separate entries
```

WDYT?

Reply:
What about that?
> Subresource is the name of the subresource used to update that object, or empty string if the object was updated through the main resource. The value of this field is used to distinguish between managers, even if they share the same name. For example, a status update will be distinct from a regular update using the same manager name.

Reply:
Much better, thanks @apelisse